### ISSUE TITLE
conda 4.1.0rc3 breaks constructor

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda#2513]
constructor works fine with conda 4.0.6, but using conda 4.1.0rc3, I get:
```
Fetching package metadata ....Traceback (most recent call last):
  File "/Users/ilan/python/bin/constructor", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/ilan/constructor/bin/constructor", line 6, in <module>
    sys.exit(main())
  File "/Users/ilan/constructor/constructor/main.py", line 146, in main
    verbose=opts.verbose)
  File "/Users/ilan/constructor/constructor/main.py", line 81, in main_build
    fcp.main(info, verbose=verbose)
  File "/Users/ilan/constructor/constructor/fcp.py", line 170, in main
    for url in info['channels']))
  File "/Users/ilan/python/lib/python2.7/site-packages/conda/fetch.py", line 276, in fetch_index
    url_s, priority = channel_urls[channel]
TypeError: 'int' object is not iterable
```
I wonder what changed?

**Comment by ilanschnell** [comment]
CC @kalefranz @csoja

**Comment by ilanschnell** [comment]
We should also test other projects which depend directly on conda against the new version: `cas-mirror`, `conda-api`, `conda-env`, `anaconda-client`, (not sure what others, maybe `anaconda-server`, `acluster`).

**Comment by kalefranz** [comment]
Oh, the "type" of `channel_urls` changed I think.  Or something closely related there.  It happened via some of @mcg1969's work to prioritize channels.

Yes, we should test all of the other dependencies you mentioned.  And if they're all calling into a common conda function, let's do our best just to keep it the same, and then refactor what we need to within conda.  Maybe have the solver stuff call a slightly different method if possible.

**Comment by mcg1969** [comment]
Grr. Given how determined we were to tell people not to import `conda` and use it as a module, we sure do it a lot. 

In this case I propose we test to see if `channel_urls` is not a `dict`; and if not, convert it internally to that function. Something like this (but I'm not at my desk so this is not copy and paste code)
```
if not isinstance(channel_urls, dict):
    channel_urls = {url:(config.canonical_channel_name(url),k) for k, url in enumerate(channel_urls)}
```

**Comment by mcg1969** [comment]
Wait, a fallback like this is already in the code. @ilanschnell is that the full traceback?

**Comment by ilanschnell** [comment]
@mcg1969 I added the last line of the traceback `TypeError: 'int' object is not iterable`.

**Comment by mcg1969** [comment]
That did it, thanks---I know the error. Simple fix, will get out ASAP.

**Comment by kalefranz** [comment]
@ilanschnell Can you verify your bug as reported here is fixed with #2513?  Assuming yes, I'm going to close this issue.
